### PR TITLE
Mark entity bindings as unsupported operation and deprecated

### LINKED ISSUE #1485: Binding a FullEntity to a GqlQuery fails
The GqlQuery.Builder allows positional and named bindings for FullEntity values. However, when I try to use those methods, the query execution fails with the below exception. Is this supposed to work? Same error occurs when using StructuredQuery.PropertyFilter. Am I misunderstanding the purpose of these methods? 

```
Exception in thread "main" com.google.cloud.datastore.DatastoreException: An entity value is not allowed
	at com.google.cloud.datastore.spi.DefaultDatastoreRpc.translate(DefaultDatastoreRpc.java:110)
	at com.google.cloud.datastore.spi.DefaultDatastoreRpc.translate(DefaultDatastoreRpc.java:96)
	at com.google.cloud.datastore.spi.DefaultDatastoreRpc.runQuery(DefaultDatastoreRpc.java:164)
	at com.google.cloud.datastore.DatastoreImpl$1.call(DatastoreImpl.java:93)
	at com.google.cloud.datastore.DatastoreImpl$1.call(DatastoreImpl.java:89)
	at com.google.cloud.RetryHelper.doRetry(RetryHelper.java:179)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:244)
	at com.google.cloud.datastore.DatastoreImpl.runQuery(DatastoreImpl.java:88)
	at com.google.cloud.datastore.QueryResultsImpl.sendRequest(QueryResultsImpl.java:73)
	at com.google.cloud.datastore.QueryResultsImpl.<init>(QueryResultsImpl.java:57)
	at com.google.cloud.datastore.DatastoreImpl.run(DatastoreImpl.java:82)
	at com.google.cloud.datastore.DatastoreImpl.run(DatastoreImpl.java:73)
Caused by: com.google.datastore.v1.client.DatastoreException: An entity value is not allowed, code=INVALID_ARGUMENT
	at com.google.datastore.v1.client.RemoteRpc.makeException(RemoteRpc.java:126)
	at com.google.datastore.v1.client.RemoteRpc.makeException(RemoteRpc.java:169)
	at com.google.datastore.v1.client.RemoteRpc.call(RemoteRpc.java:89)
	at com.google.datastore.v1.client.Datastore.runQuery(Datastore.java:108)
	at com.google.cloud.datastore.spi.DefaultDatastoreRpc.runQuery(DefaultDatastoreRpc.java:162)
```

### Sample Code 

```
Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
FullEntity<IncompleteKey> cellNumber = FullEntity.newBuilder().set("countryCode", "1").set("areaCode", "111").set("subscriberNumber", "2223333").build();
GqlQuery query = Query.newGqlQueryBuilder(Query.ResultType.ENTITY, "SELECT * FROM Contact WHERE cellNumber=@1").addBinding(cellNumber).build();
System.out.println(query);
QueryResults<Entity> results = datastore.run(query);
while (results.hasNext()) {
	System.out.println(results.next());
}

```

### PR DESCRIPTION
**Pull request by yihanzhen**
To fix #1485.

### CODE REVIEW

**Comment by hzyi-google** [comment]
PTAL @garrettjonesgoogle @pongad @wsh

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/GqlQuery.java [hunk 1]
```diff
@@ -282,9 +282,9 @@ static Binding fromPb(com.google.datastore.v1.GqlQueryParameter argPb) {
      * @param value a {@link FullEntity} object or a list of {@link FullEntity} objects that binds
      *     to a given name
      */
+    @Deprecated
     public Builder<V> setBinding(String name, FullEntity<?>... value) {
-      namedBindings.put(name, toBinding(EntityValue.MARSHALLER, Arrays.asList(value)));
-      return this;
+      throw new DatastoreException(12, "Binding entities is not supported.", "UNIMPLEMENTED");
```
Comment (pongad) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/GqlQuery.java]: Why 12? Is there a constant somewhere to say what this means?

Comment (hzyi-google) [reply google-cloud-datastore/src/main/java/com/google/cloud/datastore/GqlQuery.java]: This is from https://cloud.google.com/datastore/docs/concepts/errors, the link to canonical error code (https://github.com/google/googleapis/blob/master/google/rpc/code.proto) says the code for unimplemented is assigned 12 (but I am not sure they are the same think after a second look now)

Comment (pongad) [reply google-cloud-datastore/src/main/java/com/google/cloud/datastore/GqlQuery.java]: Oh I see. The codes are defined in a few places but should be consistent. Consider using [Status.Code](https://grpc.io/grpc-java/javadoc/io/grpc/Status.Code.html).

I think you can do
```java
Status.Code.UNIMPLEMENTED.value()
```

**Comment by hzyi-google** [comment]
PTAL @pongad

**Pull request merged**